pkg/controllers/disruption/helpers.go

```diff
@@ -111,11 +110,12 @@ func simulateScheduling(ctx context.Context, kubeClient client.Client, cluster *
 	}
 
 	results := scheduler.Solve(ctx, pods)
-	// check if the scheduling relied on an existing node that isn't ready yet, if so we fail
-	// to schedule since we want to assume that we can delete a node and its pods will immediately
-	// move to an existing node which won't occur if that node isn't ready.
 	for _, n := range results.ExistingNodes {
-		if !n.Initialized() || nodeutils.GetCondition(n.Node, v1.NodeReady).Status != v1.ConditionTrue {
+		// We consider existing nodes for scheduling. When these nodes are unmanaged, their taint logic should
+		// tell us if we can schedule to them or not; however, if these nodes are managed, we will still schedule to them
+		// even if they are still in the middle of their initialization loop. In the case of disruption, we don't want
+		// to proceed disrupting if our scheduling decision relies on nodes that haven't entered a terminal state.
+		if !n.Initialized() {
```

Comment:
The initialized logic isn't changed with respect to the NodeReady condition. I thought we were going to model initialization everywhere with the existing condition.

Reply:
Discussed offline: Nodes that aren't managed have a `kubernetes.io/not-ready` taint added to them which means that we wouldn't schedule against them. The initialized concept should _only_ be used as a Karpenter-managed node concept and it just means that the node hasn't been through its initialization loop once. Once a Karpenter node goes through its initialization loop, it is marked as initialized with a label and it can receive taints that will affect scheduling like any other node on the cluster that is unmanaged.